cli/src/command.rs

```diff
@@ -238,29 +239,32 @@ macro_rules! unwrap_client {
 			#[cfg(feature = "rococo-native")]
 			polkadot_client::Client::Rococo($client) => $code,
 			#[allow(unreachable_patterns)]
-			_ => Err(Error::CommandNotImplemented),
+			_ => {
+				let _ = $client;
+
+				Err(Error::CommandNotImplemented)
+			},
 		}
 	};
 }
 
 /// Runs performance checks.
 /// Should only be used in release build since the check would take too much time otherwise.
 fn host_perf_check() -> Result<()> {
-	#[cfg(not(build_type = "release"))]
+	#[cfg(not(feature = "hostperfcheck"))]
+	{
+		return Err(Error::FeatureNotEnabled { feature: "hostperfcheck" }.into())
+	}
+
+	#[cfg(all(not(build_type = "release"), feature = "hostperfcheck"))]
 	{
 		return Err(PerfCheckError::WrongBuildType.into())
 	}
-	#[cfg(build_type = "release")]
+
+	#[cfg(all(feature = "hostperfcheck", build_type = "release"))]
 	{
-		#[cfg(not(feature = "hostperfcheck"))]
-		{
-			return Err(PerfCheckError::FeatureNotEnabled { feature: "hostperfcheck" }.into())
-		}
-		#[cfg(feature = "hostperfcheck")]
-		{
-			crate::host_perf_check::host_perf_check()?;
-			return Ok(())
-		}
+		crate::host_perf_check::host_perf_check()?;
```

Comment:
Can we just put `#[cfg(all(feature = "hostperfcheck", build_type = "release"))]` on this function as well as on `Subcommand::HostPerfCheck`? Why have runtime checks that just return `FeatureNotEnabled`?

Reply:
Sorry, I don't get the question?

Reply:
I mean, why have this `FeatureNotEnabled` error returned at runtime, when we can just disable these code paths with conditional compilation? Then the error is not needed at all, as entering the code path is impossible.

Reply:
When executing it without the feature compiled you get a help message on what to do in order to run it.
With your suggestion a user gets something like "unknown command"